README.md

```diff
@@ -80,6 +84,35 @@ Running the federation test suite is very similar. Note: The federation test sui
 python run_federation_tests.py --tests test_federation.Test_ICommands.test_iquest__3466
 ```
 
+## Run iRODS Plugin Tests
+
+For purposes of CI, official iRODS Plugins have followed a convention of providing a "test hook" which will install the appropriate packages and run the appropriate test suite. If a test hook is provided in the prescribed way, any iRODS plugin test suite can be run in the testing environment.
+
+The test hooks generally have the following requirements:
+    - `irods_python_ci_utilities` is installed as a pip package
+    - Path to local directory with built plugin packages (passed by `--built_packages_root_directory`)
+        - Inside the root directory, the `os_specific_directory` must exist and contain the appropriate packages
+        - The `os_specific_directory` must be named like this (image tag -> directory name):
+            - ubuntu:16.04  ->  `Ubuntu_16`
+            - ubuntu:18.04  ->  `Ubuntu_18`
+            - centos:7      ->  `Centos linux_7`
+    - iRODS server is already installed and setup
+
+Your provided built packages should be in an identical directory or symlink following the naming convention above.
+
+In the future, these requirements will be relaxed so that creating a build-and-test workflow will not be as difficult.
+
+### How to run a test hook
+
+We will use the curl microservice plugin as an example. The curl microservice plugin package is assumed to have been built for the target platform. The last argument provided to the script is the name of the git repository from which the test hook will be fetched.
```

Comment:
"git repository"   ... but it's not in the example below... it's a local directory?

Reply:
The last argument is `irods_microservice_plugins_curl` which is the name of the git repository. The local directory - `/path/to/curl-microservice-plugin` - refers to the location of the built packages.

Reply:
We can hash out some of these options in the future. WIP :)